### PR TITLE
cpu/stm32f4: add stm32f413vg support

### PR DESCRIPTION
**Pull request by vincent-d**
Adds stm32f413vg support ad fixes the number of IRQ for stm32f413

### CODE REVIEW

File: cpu/stm32f4/ldscripts/stm32f413vg.ld [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * @author          Vincent Dupont <vincent@otakeys.com>
+ *
+ * @}
+ */
+
+MEMORY
+{
+    rom (rx)    : ORIGIN = 0x08000000, LENGTH = 1024K
+    ram (rwx)   : ORIGIN = 0x20000000, LENGTH = 256K
```
Comment (aabadie) [inline cpu/stm32f4/ldscripts/stm32f413vg.ld]: According to the datasheet, the values here don't seem to be valid. I think it should be:
```
rom (rx)    : ORIGIN = 0x08000000, LENGTH = 1.5M (not sure about the dot)
ram (rwx)   : ORIGIN = 0x20000000, LENGTH = 320K
cpuid (r)   : ORIGIN = 0x1fff7a10, LENGTH = 12
```
and there's no ccmram apparently.

Comment (vincent-d) [inline cpu/stm32f4/ldscripts/stm32f413vg.ld]: The VG is a 1M flash variant.

But you're right about ram, I fixed stm32f413zh as well.

By the way, I'm wondering if we should not rename stm32f413xh and stm32f413xg as there is only one vendor header for all stm32f413xx and there is only two flash sizes (G = 1M, H = 1.5M).

**Comment by aabadie** [comment]
> there is only one vendor header for all stm32f413xx and there is only two flash sizes (G = 1M, H = 1.5M)

Until when ? ;) I think it's fine like this.

**Comment by vincent-d** [comment]
> Until when ? ;) 

I think you've made a point, but it's a pain if we want to support every single part number for all families. For this one (f413) there is 5 packages and 2 memory sizes, which would lead to 10 linker scripts and a lot of `|| defined(xxxx)`.
Anyway, let's add that, we can have this discussion later on ;)

**Comment by vincent-d** [comment]
Squashed

File: cpu/stm32f4/include/cpu_conf.h [hunk 2]
```diff
@@ -50,7 +50,11 @@ extern "C" {
  * @{
  */
 #define CPU_DEFAULT_IRQ_PRIO            (1U)
+#if defined(CPU_MODEL_STM32F413ZH) || defined(CPU_MODEL_STM32F413VG)
+#define CPU_IRQ_NUMOF                   (102U)
```
Comment (haukepetersen) [inline cpu/stm32f4/include/cpu_conf.h]: You state here, that the `stm32f413xx` CPUs have 102 interrupt lines, but the vector definitions in the `vectors.c` file don`t reflect this. So it seems to me, that the `vectors.c` file need adaption as well?

Comment (vincent-d) [inline cpu/stm32f4/include/cpu_conf.h]: It seems like I've missed a lot of things!

I'm on it.

Comment (vincent-d) [inline cpu/stm32f4/include/cpu_conf.h]: Fixed. But it seems that for other part numbers, some vector definitions are wrong.

File: cpu/stm32f4/vectors.c [hunk 3]
```diff
@@ -220,20 +220,43 @@ ISR_VECTORS const void *interrupt_vector[] = {
...
     (void*) isr_otg_hs_wkup,        /* USB OTG HS Wakeup through EXTI */
     (void*) isr_otg_hs,             /* USB OTG HS */
     (void*) isr_dcmi,               /* DCMI */
-#endif
     (void*) isr_cryp,               /* CRYP crypto */
+#endif
     (void*) isr_hash_rng,           /* Hash and Rng */
     (void*) isr_fpu,                /* FPU */
+#if defined(CPU_MODEL_STM32F413ZH) || defined(CPU_MODEL_STM32F413VG)
+    (void*) isr_usart7,             /* USART7 */
```
Comment (aabadie) [inline cpu/stm32f4/vectors.c]: This functions need to be declared above: Murdock is failing right now.

**Comment by vincent-d** [comment]
> This functions need to be declared above: Murdock is failing right now.

Amended directly, sorry.

**Comment by aabadie** [comment]
Re-ACK !

**Comment by vincent-d** [comment]
@haukepetersen, did I fix all your remarks?

**Comment by vincent-d** [comment]
ping

**Comment by aabadie** [comment]
Approval from @haukepetersen still missing.

**Comment by aabadie** [comment]
ping @haukepetersen

**Comment by vincent-d** [comment]
ping @haukepetersen

**Comment by aabadie** [comment]
ping @haukepetersen, just an ACK from your side and we are good to go here.

**Comment by vincent-d** [comment]
Rebased

ping @haukepetersen, could you ACK?

**Comment by aabadie** [comment]
@haukepetersen can you ACK ?

**Comment by vincent-d** [comment]
I'd really like to have this on next release, @haukepetersen could you ACK?

**Pull request merged**